### DISCUSSION TITLE
IDE-execution protocol for /run_session seems broken with VS 17.10.0 Preview 4.0

### DISCUSSION THREAD

**Discussion by epsitec** [microsoft/aspire#3819] [Q&A] [answered]
I have been replicating what DCP does, based on https://github.com/dotnet/aspire/blob/main/docs/specs/IDE-execution.md

```cs
var port = Environment.GetEnvironmentVariable ("DEBUG_SESSION_PORT");
var token = Environment.GetEnvironmentVariable ("DEBUG_SESSION_TOKEN");
```

With the latest version of Visual Studio (Version 17.10.0 Preview 4.0) installed yesterday, my code stopped working.

I see that the API has changed. However, I must be missing something in the documentation, as all my attempts to `run_session` return a `400`. Calling `/info` works, though.

```
PUT https://localhost:53333/run_session?api-version=2024-03-03
```

with headers:

```
Authorization: Bearer xxxx
Microsoft-Developer-DCP-Instance-ID: 0123456789ab
```

and here is my payload:

```json
{
    "project_path": "S:\\git\\App.Server\\App.Server.csproj",
    "debug": true,
    "env" : [],
    "args": [],
    "launch_profile": "http"
}
```

@karolz-ms what am I doing wrong?

**Comment by davidfowl** [comment]
cc @karolz-ms

**Comment by karolz-ms** [comment, marked as answer]
@epsitec the payload for the request has changed as well https://github.com/dotnet/aspire/blob/main/docs/specs/IDE-execution.md#create-session-request It now contains an array of launch configurations (currently only one is supported, but more than one may be supported in future). The project path and launch profile appear under the launch configuration.

Also make sure you are using HTTPS when talking to the endpoint.

Hope this helps, but please do not hesitate to ping us if you have any follow-up questions.

**Comment by epsitec** [comment]
Thank you very much for the clarification. I feel stupid not having seen that the format of the payload changed too!
Everything is working fine now.
